### PR TITLE
adds .update method

### LINKED ISSUE #48: update user query
Need to create a query to update existing user

### PR DESCRIPTION
**Pull request by mk4111**
Closes #48 

Creates method, tests endpoint and query string.
Adds validator for incoming object.

### CODE REVIEW

**Comment by mk4111** [comment]
Need to add Readme and update version

**Comment by mk4111** [comment]
Hi @JMurphyWeb and @SimonLab could you have a wee review please? Thank youuu :taco:

File: lib/index.js [hunk 1]
```diff
@@ -41,6 +42,26 @@ function register (server, options, next) {
     });
   };
 
+  var peopleEdit = function (userId, updatedProfile, cb) {
+    var fields = ['first_name', 'last_name', 'job_title', 'phone']; // should add email to this too for Admins?
+
+    validatePeopleUpdate(fields, updatedProfile, function (err) {
+      if (err) {
+        return cb(null, err);
```
Comment (SimonLab) [inline lib/index.js]: The error must be the first argument of the callback

File: lib/helpers/validate-people-update.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+var Boom = require('boom');
+
+// CURRENTLY ONLY CHECKS EACH KEY HAS A VALUE >> need to be more specific?
+module.exports = function (fields, profile, cb) {
```
Comment (SimonLab) [inline lib/helpers/validate-people-update.js]: I think we can simplify this validation by using map and hasOwnProperty function. Also I don't think the validation should check that **all** fields are defined but that the fields in the object are valid field.

File: lib/index.js [hunk 3]
```diff
@@ -41,6 +42,26 @@ function register (server, options, next) {
...
+      }
+
+      return query(queries.peopleEdit(userId, fields, updatedProfile), pool, function (err, res) { //eslint-disable-line
+        if (err) {
+          return cb(err);
+        }
+        if (res.rowCount === 0) {
+          return cb(null, Boom.notFound('The user specified does not exist'));
```
Comment (SimonLab) [inline lib/index.js]: cb(error, result) see comment above, the error must be the first argument

File: tests/people/edit/query.test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+var test = require('tape');
+var query = require('../../../lib/queries/people/edit.js');
+
+var fields = ['first_name', 'last_name', 'job_title', 'phone']; // should add email to this too for Admins?
```
Comment (SimonLab) [inline tests/people/edit/query.test.js]: It's better to add the questions on the issue. The plugin is independent to the main application so we don't have any admins or primary. The role of the function is to update some fields. Then it's the responsibility of the code where you call this function to pass the right fields.

File: example/server.js [hunk 5]
```diff
@@ -123,7 +123,25 @@ function init (config, callback) {
...
             method: 'POST',
+            path: '/peopleEdit',
+            handler: function (request, reply) {
+              var id = request.query.id;
+              var profileUpdate = request.payload;
+
+              request.server.methods.pg.people.edit(id, profileUpdate, function (error, response) { // eslint-disable-line
+                Hoek.assert(!error, 'people.edit error');
+
+                if (response.rowCount === 0) {
```
Comment (jay-meister) [inline example/server.js]: thnk we need to remove this

File: tests/people/edit/endpoint.test.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+test('Edit user profile', function (t) {
+  init(config, function (err, server, pool) {
+    if (err) {
+      console.log('error initialise server', err);
+      return t.fail();
+    }
+    server.inject(editProfile(3), function (res) {
+      t.equal(res.statusCode, 200, 'User not found, 404 returned');
```
Comment (jay-meister) [inline tests/people/edit/endpoint.test.js]: message needs updating

**Review by jay-meister — APPROVED** [review]
Niiiiiiiice

**Review by SimonLab — APPROVED** [review]
Thanks @mk4111 @JMurphyWeb

**Pull request merged**